packages/mui-system/src/sizing.js

```diff
@@ -16,8 +16,21 @@ export const maxWidth = (props) => {
     const styleFromPropValue = (propValue) => {
       const breakpoint =
         props.theme?.breakpoints?.values?.[propValue] || breakpointsValues[propValue];
+
+      if (!breakpoint) {
+        return {
+          maxWidth: sizingTransform(propValue),
+        };
+      }
+
+      if (props.theme?.breakpoints?.unit !== 'px') {
```

Comment:
Nit: Do we need to check if it's not `px`? Why not just append the unit?

Reply:
@brijeshb42 I just didn't want to change the existing behavior for pixels - i.e. returning `maxWidth: 100` instead of `maxWidth: '100px'` 😬